packages/eas-cli/src/graphql/queries/StatuspageServiceQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { print } from 'graphql';
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../client';
+import {
+  StatuspageServiceByServiceNamesQuery,
+  StatuspageServiceByServiceNamesQueryVariables,
+  StatuspageServiceFragment,
+  StatuspageServiceName,
+} from '../generated';
+import { StatuspageServiceFragmentNode } from '../types/StatuspageService';
+
+export const StatuspageServiceQuery = {
+  async statuspageServicesAsync(
+    serviceNames: StatuspageServiceName[]
+  ): Promise<StatuspageServiceFragment[]> {
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .query<StatuspageServiceByServiceNamesQuery, StatuspageServiceByServiceNamesQueryVariables>(
+          gql`
+            query StatuspageServiceByServiceNamesQuery($serviceNames: [StatuspageServiceName!]!) {
+              statuspageService {
+                byServiceNames(serviceNames: $serviceNames) {
+                  id
+                  ...StatuspageServiceFragment
+                }
+              }
+            }
+            ${print(StatuspageServiceFragmentNode)}
+          `,
+          { serviceNames },
+          {
+            additionalTypenames: ['StatuspageService', 'StatuspageIncident'],
+          }
+        )
+        .toPromise()
+    );
+
+    return data.statuspageService.byServiceNames;
```

Comment:
does this not need `data?.`

Reply:
```
export async function withErrorHandlingAsync<T>(promise: Promise<OperationResult<T>>): Promise<T> {
  const { data, error } = await promise;

  if (error) {
    if (error.graphQLErrors.some(e => e?.extensions?.isTransient)) {
      Log.error(`We've encountered a transient error. Try again shortly.`);
    }
    throw error;
  }

  // Check for malformed response. This only checks the root query existence,
  // It doesn't affect returning responses with empty resultset.
  if (!data) {
    throw new Error('Returned query result data is null!');
  }

  return data;
}
```
I think that it doesn't because of how `withErrorHandlingAsync` works